src/nvim/api/vim.c

```diff
@@ -164,6 +165,39 @@ Object vim_eval(String str, Error *err)
   return rv;
 }
 
+/// Call the given function with the given arguments stored in an array.
+///
+/// @param functionName The function to call
+/// @param args the functions arguments packed in an Array
+/// @param[out] err Details of an error that may have occurred
+/// @return The result of the function call if it was successful
+Object vim_function(String functionName, Array args, Error *err)
+  FUNC_ATTR_DEFERRED
+{
+  Object rv = OBJECT_INIT;
+  typval_T vim_args[args.size+1];
```

Comment:
By the way, you *must* check that `args.size <= MAX_FUNC_ARGS`. There is a hard limit here, you will get a crash if you exceed it, especially for user functions. Thus you don’t need VLA, just have a statically sized array with MAX_FUNC_ARGS+1 like func_call has.